app/workers/carto_db_importer.rb

```diff
@@ -32,10 +32,11 @@ def self.perform(data_layer_id)
       import_status = CartoDb.import_status(queue_id)
       i += 1
     end
-    layer.update_attributes(import_status: import_status["state"])
-    layer.update_attributes(table_name: import_status["table_name"])
+    layer.table_name = import_status["table_name"]
```

Comment:
the update_attributes would write to the table each time, so I've just attributed and saved at the end.